### PR TITLE
Add `runtime.model.Model` class to save/load SQLFlow model

### PR DESCRIPTION
**Pull request by Yancey0623**
Implement `runtime.model.Model` class.
TODO: to save SQLFlow model with sqlfs and local fs.
 
c.f. #2667

### CODE REVIEW

File: python/runtime/model/model.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This module saves or loads the SQLFlow model.
+"""
+import os
+from enum import Enum
+
+# TODO(yancey1989): move diag to the upper direction
```
Comment (typhoonzero) [inline python/runtime/model/model.py]: direction -> directory?

Comment (typhoonzero) [reply python/runtime/model/model.py]: The name `diag` is confusing, maybe use `diagnostic.py`?

File: python/runtime/model/model.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        uri, string
+            model saving URL with format: <driver>://<path>
+
+        datasource, string
+            the connection datasource DSN is required if saving with sqlfs.
+        """
+        driver, path = parseURI(uri)
+        if driver == "file":
+            # TODO(yancey1989): save model with local file system.
+            pass
```
Comment (typhoonzero) [inline python/runtime/model/model.py]: raise NotImplementedError

File: python/runtime/model/model.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        elif driver == "sqlfs":
+            # TODO(yancey1989): save model with SQL file system.
+            pass
```
Comment (typhoonzero) [inline python/runtime/model/model.py]: raise NotImplementedError

File: python/runtime/model/model.py (hunk 1 above)
Comment (Yancey0623) [reply python/runtime/model/model.py]: > The name diag is confusing, maybe use diagnostic.py?

I will do that in the next PR.

File: python/runtime/model/model.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+                                          model_params={...})
+    m = runtime.model.Model(ModelType.XGBOOST, meta)
+    m.save(uri="sqlfs://mydatabase/mytable")
+
+    """
+    def __init__(self, typ, meta):
+        """
+        Args:
+
+        typ, ModelType
```
Comment (lhw362950217) [inline python/runtime/model/model.py]: typ, ModelType ->         typ: ModelType

File: python/runtime/model/model.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            pass
+        else:
+            raise SQLFlowDiagnostic("unsupported driven to save model: %s" %
+                                    driver)
+
+
+def parseURI(uri):
+    """Parse the model URI into two parts: driver and path.
+    """
+    array = uri.split("://")
```
Comment (lhw362950217) [inline python/runtime/model/model.py:87]: What does sqlfs schema look like?  sqlfs://mysql://xxx or mysql://xxx, it seem not compatible with `save` function.

File: python/runtime/model/model.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+class EstimatorType(Enum):
+    """The enum type for various SQLFlow estimator.
+    """
+    TENSORFLOW = 1
+    XGBOOST = 2
+
+
+class Model(object):
+    """Model module represents a SQLFlow trained model, which includes
+three parts:
```
Comment (shendiaomo) [inline python/runtime/model/model.py]: indentation

File: python/runtime/model/model.py (hunk 5 above)
Comment (Yancey0623) [reply python/runtime/model/model.py:87]: It's like `sqlfs://user/yanxu`, `mysq://xxx` is the datasource DNS format, and we can pass it to `model.save(uri="sqlfs://xxx", datasource="mysql://")`.

**Review by typhoonzero — APPROVED** [review]

**Review by lhw362950217 — APPROVED** [review]
LGTM

**Pull request merged**